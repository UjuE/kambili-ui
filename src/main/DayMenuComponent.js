import React from 'react';
import DayComponent from "./DayComponent"
import MealComponent from "./MealComponents";
import moment from 'moment'

export default class DayMenuComponent extends React.Component {
    render() {
        var day = this.props.day;
        var mealComponent =
            this.props.meals.map((x) => {
                return (
                    <MealComponent key={day + x.meal_type} type={x.meal_type} name={x.meal_name} />
                )
            });

        var isToday = moment().format('dddd').toLocaleLowerCase() === this.props.day.toLocaleLowerCase() ? "today" : "";
        var className = "day-block "+ isToday;
        return (
            <div className={className}>
                <DayComponent value={day}/>
                {mealComponent}
            </div>
        )
    }
}