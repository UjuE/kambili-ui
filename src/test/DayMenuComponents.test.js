import React from 'react';
import { shallow } from 'enzyme';
import DayMenuComponent from '../main/DayMenuComponent';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

describe("DayMenuComponent", () => {
    it("should accept parameters", () => {
        const component = shallow(<DayMenuComponent day="Sunday"
                                                    breakfast="Bread and Butter"
                                                    dinner="Garri with assorted meat banga soup"
                                                    lunch="Garri with assorted meat banga soup" />);
        expect(component.getElements()).toMatchSnapshot();
    });
});