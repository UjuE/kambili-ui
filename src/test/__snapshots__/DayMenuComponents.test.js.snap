// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DayMenuComponent should accept parameters 1`] = `
Array [
  <div
    className="day-block"
  >
    <DayComponent
      value="Sunday"
    />
    <MealComponent
      name="Food"
      type="breakfast"
    />
    <MealComponent
      name="Food"
      type="lunch"
    />
  </div>,
]
`;
